LanguageFeatures/Augmentation-libraries/augmented_expression_A11_t19.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The exact result of an `augmented` expression depends on what is
+/// being augmented, but it generally follows the same rules as any normal
+/// identifier:
+/// ...
+/// - Augmenting non-redirecting generative constructors: Unlike other
+///   functions, `augmented` has no special meaning in non-redirecting
+///   generative constructors. It is still a reserved word inside the body of
+///   these constructors, since they are within the scope of an augmenting
+///   declaration.
+///
+/// @description Checks that it is a compile-time error for an augmenting
+/// non-redirecting generative constructor to have as a named parameter a record
+/// with a named field with the name `augmented`.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+class C {
+  C({({int augmented}) r = (augmented: 0)});
+  C.foo({({int augmented}) r = (augmented: 0)});
+}
+
+augment class C {
+  augment C({({int augmented}) r}) {}
+//                 ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  augment C.foo({({int augmented}) r}) {}
+//                     ^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  C.bar({({int augmented}) r = (augmented: 0)}) {}
+//             ^^^^^^^^^
```

Comment:
Can we have `^^^^^^^^^^^^` for both occurrences?

Reply:
Yes, we can but in this case we should use exact error code/error message. Like
```
// ... some code ...
// ^
// [analyzer] COMPILE_TIME_ERROR.UNDEFINED_GETTER
//                 ^^^^^^^^^^^
// [analyzer] STATIC_WARNING.UNNECESSARY_NON_NULL_ASSERTION
```
In case of `unspecified` we should have only one error position (which is ignored). We can't have two error in this case. In both cases below the test will fail.
```dart
//   ...code....
//        ^^^^    ^^^^^
// [analyzer] unspecified
// [cfe] unspecified

//   ...code...
//        ^^^^ 
// [analyzer] unspecified
// [cfe] unspecified
//                ^^^^^
// [analyzer] unspecified
// [cfe] unspecified
```

Reply:
OK!